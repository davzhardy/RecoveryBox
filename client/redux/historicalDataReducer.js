const initialState = {

  1600300800000: {
    date: 1600300800000,
    meetings: 2,
    feeling: 5,
    moods: ['tired', 'happy', 'bored'],
    suggestions: ['Meetings', 'Prayer', 'Meditation', 'Useful', 'Kind to Myself', 'Gratitude', 'Step 10s', 'Inventory', 'Called a Newcomer'],
  },
  1600387200000: {
    date: 1600387200000,
    meetings: 1,
    feeling: 4,
    moods: ['tired', 'happy', 'down'],
    suggestions: ['Gratitude', 'Prayer', 'Meditation', 'Useful', 'Kind to Myself']
  },
  1600473600000: {
    date: 1600473600000,
    meetings: 0,
    feeling: 3,
    moods: ['elated', 'happy', 'sad'],
    suggestions: ['Inventory', 'Called a Newcomer', 'Prayer', 'Meditation', 'Useful', 'Kind to Myself']
  },
  1600560000000: {
    date: 1600560000000,
    meetings: 2,
    feeling: 2,
    moods: ['joyful', 'happy', 'sad'],
    suggestions: ['Kind to Myself']
  },
  1600646400000: {
    date: 1600646400000,
    meetings: 5,
    feeling: 5,
    moods: ['bored', 'happy', 'sad'],
    suggestions: ['Prayer', 'Meditation', 'Useful', 'Kind to Myself']
  },
  1600819200000: {
    date: 1600819200000,
    meetings: 2,
    feeling: 10,
    moods: [],
    suggestions: []
  },
  1600905600000: {
    date: 1600905600000,
    meetings: 0,
    feeling: 9,
    moods: ['stressed', 'happy', 'sad'],
    suggestions: ['Prayer', 'Meditation', 'Useful', 'Kind to Myself']
  },
  1601078400000: {
    date: 1601078400000,
    meetings: 5,
    feeling: 7,
    moods: ['tired', 'happy', 'sad'],
    suggestions: ['Prayer', 'Meditation', 'Useful', 'Kind to Myself']
  }
}

// TODO refactor this to some kind of array so the data is more easily accessible

const historicalDataReducer = (state = initialState, action) => {
  switch (action.type) {
    case "UPDATE_HISTORICALDATA_WITH_DAILYINFO":
      return {...state, key: action.payload}
    case "UPDATE_HISTORICALDATA_WITH_HISTORICALINFO":
      return {...state, newkey: action.payload}
    case "CREATE_HISTORICALDATA":
      return {...state, ...action.payload}
    default:
      return state;
  }
}

export default historicalDataReducer;


// TODO delete this historical data when im 100% sure I dont need it 
 // 1600300800000: {
  //   date: 1600300800000,
  //   meetings: 2,
  //   feeling: 5,
  //   moods: ['tired', 'happy', 'bored'],
  //   suggestions: ['Meetings', 'Prayer', 'Meditation', 'Useful', 'Kind to Myself', 'Gratitude', 'Step 10s', 'Inventory', 'Called a Newcomer'],
  // },
  // 1600387200000: {
  //   date: 1600387200000,
  //   meetings: 1,
  //   feeling: 4,
  //   moods: ['tired', 'happy', 'down'],
  //   suggestions: ['Gratitude', 'Prayer', 'Meditation', 'Useful', 'Kind to Myself']
  // },
  // 1600473600000: {
  //   date: 1600473600000,
  //   meetings: 0,
  //   feeling: 3,
  //   moods: ['elated', 'happy', 'sad'],
  //   suggestions: ['Inventory', 'Called a Newcomer', 'Prayer', 'Meditation', 'Useful', 'Kind to Myself']
  // },
  // 1600560000000: {
  //   date: 1600560000000,
  //   meetings: 2,
  //   feeling: 2,
  //   moods: ['joyful', 'happy', 'sad'],
  //   suggestions: ['Kind to Myself']
  // },
  // 1600646400000: {
  //   date: 1600646400000,
  //   meetings: 5,
  //   feeling: 5,
  //   moods: ['bored', 'happy', 'sad'],
  //   suggestions: ['Prayer', 'Meditation', 'Useful', 'Kind to Myself']
  // },
  // 1600819200000: {
  //   date: 1600819200000,
  //   meetings: 2,
  //   feeling: 10,
  //   moods: [],
  //   suggestions: []
  // },
  // 1600905600000: {
  //   date: 1600905600000,
  //   meetings: 0,
  //   feeling: 9,
  //   moods: ['stressed', 'happy', 'sad'],
  //   suggestions: ['Prayer', 'Meditation', 'Useful', 'Kind to Myself']
  // },
  // 1601078400000: {
  //   date: 1601078400000,
  //   meetings: 5,
  //   feeling: 7,
  //   moods: ['tired', 'happy', 'sad'],
  //   suggestions: ['Prayer', 'Meditation', 'Useful', 'Kind to Myself']
  // },