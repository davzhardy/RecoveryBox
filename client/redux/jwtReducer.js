const initialState = {
	jwt: '' ,
}

const jwtReducer = (state = initialState, action) => {
	switch (action.type) {
		case 'SAVE_JWT': 
		return {
			...state,
			jwt: action.payload
		}
		case 'DELETE_JWT': 
		return {
			...state,
			jwt: '',
		}
		default:
      return state;
	}
}

export default jwtReducer;