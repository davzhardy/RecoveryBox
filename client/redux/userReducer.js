const initialState = {
  id: false,
  email: false,
  username: false,
  password: false,
  firstName: false,
  lastName: false,
  registrationDate: false,
}

const userReducer = (state = initialState, action) => {
  switch (action.type) {
    case "UPDATE_USERNAME":
      return {
        ...state,
        username: action.payload
      }
    case "UPDATE_USERINFO":
      return {
        ...state,
        id: action.payload.id,
        email: action.payload.email,
        username: action.payload.username,
        firstName: action.payload.firstName,
        lastName: action.payload.lastName,
        registrationDate: action.payload.registrationDate,
      }
    default:
      return state;
  }
}

export default userReducer;