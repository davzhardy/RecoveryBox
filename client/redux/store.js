import { createStore } from "redux";
import { combineReducers } from "redux";
import dailyInfoReducer from './dailyInfoReducer'
import settingsReducer from './settingsReducer'
import userReducer from './userReducer'
import quoteReducer from './quoteReducer'
import historicalDataReducer from './historicalDataReducer'
import helperReducer from './helperReducer'
import calendarHistoricalDayReducer from './calendarHistoricalDayReducer'
import jwtReducer from './jwtReducer';

const rootReducer = (combineReducers)({
  user: userReducer,
  dailyInfo: dailyInfoReducer,
  settings: settingsReducer,
  dailyQuote: quoteReducer,
  historicalData: historicalDataReducer,
  helper: helperReducer,
  calendarHistoricalDay: calendarHistoricalDayReducer,
  jwt: jwtReducer,
})

const store = createStore(rootReducer);

export default store
