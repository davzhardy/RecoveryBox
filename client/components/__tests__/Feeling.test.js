import React from 'react';
import { Provider } from "react-redux";
import { createStore } from "redux";
import renderer from 'react-test-renderer';

import dailyInfoReducer from '../../redux/dailyInfoReducer'
import ModifySuggestionItem from '../Feeling';

const initialState = {
	dailyInfo: {
		meetings: 0,
		feeling: 0,
		moods: [],
		suggestions: [],
	}
}

const store = createStore(dailyInfoReducer, initialState);

test('renders correctly', () => {
	const tree = renderer.create(<Provider store = {store}><ModifySuggestionItem/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});