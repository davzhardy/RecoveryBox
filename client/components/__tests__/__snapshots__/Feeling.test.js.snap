// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#2A2A30",
          "fontFamily": "Montserrat_700Bold",
          "fontSize": 17,
          "fontStyle": "normal",
          "marginBottom": 10,
        }
      }
    >
      How are you feeling?
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#7B7C86",
            "fontFamily": "Montserrat_500Medium",
            "fontSize": 12,
            "fontStyle": "normal",
            "marginBottom": 10,
          }
        }
      >
        Its been a tough day
      </Text>
    </View>
    <RNCSlider
      disabled={false}
      enabled={true}
      inverted={false}
      maximumTrackTintColor="#DBDBDB"
      maximumValue={10}
      minimumTrackTintColor="#2F80ED"
      minimumValue={0}
      onChange={[Function]}
      onRNCSliderSlidingComplete={[Function]}
      onRNCSliderSlidingStart={null}
      onRNCSliderValueChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      step={1}
      style={
        Array [
          Object {
            "height": 40,
          },
          Object {
            "height": 40,
            "width": 200,
          },
        ]
      }
      thumbImage={null}
      thumbTintColor="#FF8787"
      value={5}
    />
    <View
      style={
        Object {
          "width": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#7B7C86",
            "fontFamily": "Montserrat_500Medium",
            "fontSize": 12,
            "fontStyle": "normal",
            "marginBottom": 10,
            "textAlign": "left",
          }
        }
      >
        Super awesome
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": -15,
        "width": 150,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/sad.png",
        }
      }
      style={
        Object {
          "height": 30,
          "opacity": 0,
          "width": 30,
        }
      }
    />
    <Image
      source={
        Object {
          "testUri": "../../../assets/neutral.png",
        }
      }
      style={
        Object {
          "height": 30,
          "opacity": 1,
          "width": 30,
        }
      }
    />
    <Image
      source={
        Object {
          "testUri": "../../../assets/happy.png",
        }
      }
      style={
        Object {
          "height": 30,
          "opacity": 0,
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "height": 1,
        "width": "80%",
      }
    }
  />
</View>
`;
